src/document/mod.rs

```diff
@@ -235,85 +237,81 @@ impl DocumentOps for crate::IronOxide {
                 }
             };
 
-        self.runtime.block_on(document_api::encrypt_document(
+        document_api::encrypt_document(
             self.device.auth(),
             &self.recrypt,
             &self.user_master_pub_key,
             &self.rng,
             document_data,
-            encrypt_opts.id,
-            encrypt_opts.name,
+            maybe_doc_id,
+            maybe_doc_name,
```

Comment:
What's the reasoning behind this change? To me it looks the same but with an extra `clone()` thrown onto the document id and name

Reply:
I was getting compiler errors about partial moves. This was my way around that.